utils/src/range/mod.rs

```diff
@@ -146,6 +151,20 @@ where
     }
 }
 
+impl<T: Copy + Ord> TryFrom<RangeSet<T>> for Range<T> {
+    type Error = RangeSet<T>;
+
+    /// Attempts to convert a `RangeSet` into a single `Range`, returning the set if it
+    /// does not contain exactly one range.
+    fn try_from(set: RangeSet<T>) -> Result<Self, Self::Error> {
+        if set.len_ranges() == 1 {
+            Ok(set.ranges.into_iter().next().unwrap())
+        } else {
+            Err(set)
```

Comment:
Is that a convention to return the input as an error variant if `try_from` fails? I find it a bit strange.